pallene/to_ir.lua

```diff
@@ -82,10 +82,60 @@ function ToIR:convert_stat(cmds, stat)
         local body = {}
         self:convert_stat(body, stat.block)
 
-        table.insert(cmds, ir.Cmd.For(stat.loc,
+        table.insert(cmds, ir.Cmd.ForNum(stat.loc,
             v, start, limit, step,
             ir.Cmd.Seq(body)))
 
+    elseif tag == "ast.Stat.ForIn" then
+        assert(stat.index._name._tag == "checker.Name.Local")
+        assert(stat.decl._name._tag == "checker.Name.Local")
+
+        local arr = stat.exp.args[1].exp
+        local index = ir.Value.LocalVar(stat.index._name.id)
+        local elem = ir.Value.LocalVar(stat.decl._name.id)
+        local arrVar = self:exp_to_value(cmds, arr)
+
+        local xsId
+        if arr._type._tag == "types.T.Array" then
+            xsId = ir.add_local(self.func, false, types.T.Array(arr._type.elem))
+        end
```

Comment:
We might want to turn this if statement into an `assert`, since the code after this if statement does not work if `xsId` is not  initialized.

Reply:
done

Reply:
I think the change might not have been pushed to github yet.